src/fhir/models/encounter.ts

```diff
@@ -63,10 +63,42 @@ export class EncounterModel extends FHIRModel<fhir4.Encounter> {
   }
 
   get typeDisplay(): string | undefined {
-    const codeableConcept = find(this.resource.type, {
-      coding: [{ system: SYSTEM_ACT_CODE }],
-    });
-    const coding = findCoding(SYSTEM_ACT_CODE, codeableConcept);
-    return coding?.display || coding?.code;
+    const { display: classDisplay, system: classSystem } = this.resource.class;
+    const classCode = (this.resource.class.code ?? "").toUpperCase();
+    const firstTypeCoding = this.resource.type?.[0]?.coding?.[0] ?? {};
+    const { text: typeText } = this.resource.type?.[0] ?? {};
+    const typeCode = (firstTypeCoding.code ?? "").toUpperCase();
+
+    // Grab our code and display, first from class if valid, otherwise from
+    // first type coding.
+    let code;
+    let display;
+    if (classSystem !== SYSTEM_NULL_FLAVOR) {
+      code = classCode;
+      display = classDisplay;
+    } else {
+      code = typeCode;
+      display = typeText;
+    }
```

Comment:
nonblocking, just an option to reduce the body of function:
```suggestion
    const isNullFlavor = classSystem === SYSTEM_NULL_FLAVOR;
    const code = isNullFlavor ? typeCode : classCode;
    const display = isNullFlavor ? typeText : classDisplay;
```

Reply:
Done! Thanks for the suggestion!